server/controllers/applications/addHiredMember.js

```diff
@@ -1,3 +1,26 @@
-module.exports = (req, res) => {
-  res.send('new hire member');
+const yup = require('yup');
+const { addHiredMember } = require('../../database/queries/applications/index');
+
+module.exports = (req, res, next) => {
+  const { member_id: memberId, offer_id: offerId, status } = req.body;
+  const schema = yup.object({
+    offerId: yup.string().required(),
+    memberId: yup.string().required(),
+    status: yup.string().required(),
+  });
+  schema
+    .validate({ offerId, memberId, status })
+    .then(() => {
+      addHiredMember(memberId, offerId, status)
+        .then(() => {
+          res.send({
+            error: null,
+            data: [req.body],
```

Comment:
@AymanAlqoqa You could add `RETURNING *` into your query below, and then here you could have the endpoint send that data in the response:
```
.then( result => {
  res.send({
    error: null,
    data: [result],
  }
```

Then your frontend will get the entire inserted object, instead of just the post body, which could give back helpful additional data (for example the inserted id, or any extra database fields that had default values)

Reply:
I know it is better to return the inserted data, but I think no need for this simple situation.
I'll do it.
thanks

Reply:
Yeah you're probably right, in this situation.  Usually it's good to return the inserted ID to the client because the client will then redirect the user to a frontend route that uses that ID (for example, after creating an application the frontend might route to /applications/1) but for this situation, that probably won't happen. 🤷‍♂️

Reply:
:100: